components/tidb_query_datatype/src/codec/mysql/duration.rs

```diff
@@ -429,14 +433,25 @@ impl Duration {
     /// See: <http://dev.mysql.com/doc/refman/5.7/en/fractional-seconds.html>
     pub fn parse(ctx: &mut EvalContext, input: &str, fsp: i8) -> Result<Duration> {
         let fsp = check_fsp(fsp)?;
-        parser::parse(ctx, input, fsp, true)
+        parser::parse(ctx, input, fsp, true, false)
+            .ok_or_else(|| Error::truncated_wrong_val("TIME", input))
+    }
+
+    pub fn parse2(
```

Comment:
```suggestion
    pub fn parse_consider_overflow(
```

Why not merge it with `parse`?

Reply:
this will break a lot of test code. If rust has parameter default value like python, Ithink it will be the best way, but not. And also no override. So I just create another function.